hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -177,6 +179,9 @@ def test_function(self, data):
 
         if data.status == Status.INTERESTING:
             key = data.interesting_origin
+            if key not in self.interesting_examples:
+                self.last_bug_calls = self.call_count
+                self.stop_time = min(self.stop_time, benchmark_time() + 30)
```

Comment:
I would rather not have our generation depend on time TBH. I know it's mostly unproblematic in this case, but it makes me uncomfortable. 30 seconds is also a reasonably long time to keep the user waiting if we've already found a bug.

How would you feel about using the call count as a clock instead? i.e. if it took us N test cases to find the first bug, we keep running for at most N more test cases.

Reply:
We have two conditions based on call count already:

- run at least ten test cases, then stop at twice as many calls as it took to find the *last* bug
- never run more than 1000 calls after finding the first bug, no matter how many more we find

So the time-based cap is just to stop *really, really slow* tests in a reasonable timeframe.  I think it's worth keeping this for the rare case of extremely slow executions, which would otherwise take even longer.  If it makes you more comfortable we could reduce it to 10 seconds though.  All of this is already in a context where the user has turned the expected runtime way way up from the defaults; the longest possible runtime by default has the test pass in 20 seconds and we never generate more than `max_examples` test cases.

Honestly the main thing this buys us is not stopping at the first trivial bug or two; and that's best for attracting and teaching new users - for more complex multibug scenarios we'll still need the true fuzzing mode.